articles/express/index.md

```diff
@@ -0,0 +1,318 @@
+---
+layout: engineering-education
+status: publish
+published: true
+url: /engineering-education/express/
+title: Introduction to Express.js
+description: Express is a lightweight framework on top of Node.js, it adds functionality like middleware, template engines, and routing.
+author: rohan-reddy
+date: 2020-09-18T00:00:00-11:00
+topics: [Node.js]
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/express/hero.jpg
+    alt: expressjs example text image
+---
+[Node.js](http://nodejs.org/) is a JavaScript run time environment which is used to create server-side applications and tools. Node.js is fast, portable and written in JavaScript. Node.js does not directly support common tasks such as handling requests, serving files, handling HTTP verbs (`GET`, `POST`, …), this is where Node.js rich ecosystem comes to our aid.
+<!--more-->
+
+**Express.js (Express) is a light web framework which sits on top of Node.js and it adds functionality ([middleware](https://expressjs.com/en/guide/writing-middleware.html) , [routing](https://en.wikipedia.org/wiki/Routing), etc.) and simplicity to Node.js.**
+
+When creating a Node.js web application, we write a single JavaScript application which listens to requests from the browser, based on the request the function will send back some data or an HTML web page.
+
+![flow of a request](/engineering-education/express/node.png)
+
+[Image Credit](https://www.manning.com/books/express-in-action)
+
+A *request handler* is a JavaScript function which takes a request and sends an appropriate response.
+
+Node.js APIs can get complex and writing how to handle a single request can end up being to 50 lines of code. Express makes it easier to write web applications with Node.js.
+
+![flow with express](/engineering-education/express/express.png)
+
+[Image Credit](https://www.manning.com/books/express-in-action)
+
+### Advantages of using Express with Node.js
+- Express lets you take away a lot of Node.js complexities while adding helpful functions to a Node.js HTTP server.
+- Instead of a large request handler function, Express allows us to handle requests by writing many small modular and maintainable functions.
+- Express is *not opinionated*, meaning Express does not enforce any "right way" of doing things. You can use any compatible middleware, you can structure the app as you wish making it flexible.
+- We can integrate with a [view rendering engine](https://www.digitalocean.com/community/tutorials/nodejs-express-template-engines) of our choice like Jade, Pug, EJS, etc. A template engine enables you to use static template files and at runtime change the values of variables in those files.
```

Comment:
You should explain that a view rendering engine and template engine are one and the same. Beginners may be confused.